Utilities/priority_queue.py

```diff
@@ -12,6 +12,8 @@ class PriorityQueue():
         The priority queue.
     """
 
+    count = 987654
```

Comment:
Can you justify this for me

Reply:
I have realize that I did not choose this number correctly. A better one would be 900000. Here is the justification: this implementation of the priority queue works by appending the `priority.value` (and enum value) to the `count` variable, after which the `count` in incremented. In this way, instructions of equal priority are executed in the order that they were put into the queue.

For example, say we have two move instructions of `MEDIUM` priority. The first one gets a value of `'2' + '900000'` or `2900000` and the second one would get `'2' + '900001'` or `2900001`  associated with it. Since the queue is a min heap, the first movement would be read first, just like we want.

By choosing 900000 we could have up to 100000 instructions in the queue that will be executed in the right order. This is more than enough.

Reply:
understood, you should definitely look into writing a quick priority heap down the line, it wouldn't be a difficult change to accomplish this.